### PR TITLE
Add return field to documentation of 'get_major_ticks'

### PR DESCRIPTION
**Pull request by vishalBindal**
## PR Summary 
The documentation of axis.Axis.get_major_ticks is updated to list its return type, which is a list of Tick objects. Reference is added to the Tick object documentation. See #17015 

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

File: lib/matplotlib/axis.py [hunk 1]
```diff
@@ -1320,7 +1320,13 @@ def get_minor_formatter(self):
         return self.minor.formatter
 
     def get_major_ticks(self, numticks=None):
-        """Get the tick instances; grow as necessary."""
+        """
+        Get the tick instances; grow as necessary.
+        
+        Returns
+        -------
+        list of `.Tick` objects
```
Comment (timhoffm) [inline lib/matplotlib/axis.py:1328]:
```suggestion
        list of `.Tick`
```
We also just say "list of str" not "list of str objects".

File: lib/matplotlib/axis.py [hunk 2]
```diff
@@ -1320,7 +1320,13 @@ def get_minor_formatter(self):
         return self.minor.formatter
 
     def get_major_ticks(self, numticks=None):
-        """Get the tick instances; grow as necessary."""
+        """
+        Get the tick instances; grow as necessary.
```
Comment (timhoffm) [inline lib/matplotlib/axis.py:1324]: The "grow as necessary" part is an implementation detail which is irrelevant for the the user and rather confusing. I'd remove that. Then we can simply

~~~
r"""Return the list of major `.Tick`\s."""
~~~

**Comment by vishalBindal** [comment]
@timhoffm I've made the change. Please review

**Comment by timhoffm** [comment]
Thanks, and congratulations on your first contribution to Matplotlib. :+1:

**Comment by vishalBindal** [comment]
Thank you! :)

**Pull request merged**